pyresample/spherical.py

```diff
@@ -253,12 +254,16 @@ def intersections(self, other_arc):
         From http://williams.best.vwh.net/intersect.htm
         """
         if self.end.lon - self.start.lon > np.pi:
+            self = copy.deepcopy(self)
```

Comment:
I don't know if one should copy here, but calling a copy of self `self` may be confusing (I've never seen it before), maybe you can give a different name to the copy?  `new = copy.deepcopy(self)` maybe?

Reply:
I tried to mention this in the issue you commented on @ghiggi, but I don't think you need `self` here at all. If you start out this method with gather the lon/lat values of each arc point then `self` and `other_arc` ~can remain untouched as they aren't passed to anything after this~. Ah I see they are used in cross2cart, but I'm not sure that matters as it is the start SCoordinate that is used

```python
# these are scalars, right?
end_lon = self.end.lon
other_end_lon = other_arc.end.lon

... if statements that modify end_lon and other_end_lon ...

ea_ = self.start.cross2cart(SCoordinate(end_lon, self.end.lat)).normalize()
eb_ = other_arc.start.cross2cart(SCoordinate(other_end_lon, other_arc.end.lat)).normalize()
```